sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetQuerySuite.scala

```diff
@@ -229,4 +229,81 @@ class ParquetQuerySuite extends QueryTest with ParquetTest with SharedSQLContext
       }
     }
   }
+
+  test("SPARK-10301 Clipping nested structs in requested schema") {
```

Comment:
I added all these test cases in https://github.com/liancheng/spark/commit/cd91a26c86c72f25221ca947c5486a910194a751 of PR #8583. The 2nd case helped spotting a parquet-mr bug.